### PR TITLE
Asset editor

### PR DESCRIPTION
**Pull request by tomekpaszek**
Initial version of the Input Asset editor

### CODE REVIEW

File: Packages/com.unity.inputsystem/InputSystem/Editor/InputActionAsset/CopyPasteUtility.cs [hunk 1]
```diff
@@ -0,0 +1,164 @@
+﻿using System;
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/InputSystem/Editor/InputActionAsset/CopyPasteUtility.cs]: Everything that is editor-specific needs to have `#if UNITY_EDITOR ... #endif` around it. There's no automatic exclusion of editor stuff based on folder names in packages.

File: Packages/com.unity.inputsystem/InputSystem/Editor/InputActionAsset/CopyPasteUtility.cs [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+using System.Linq;
+using System.Text;
+using UnityEditor;
+
+namespace UnityEngine.Experimental.Input.Editor
+{
+    class CopyPasteUtility
+    {
+        const string m_InputAssetMarker = "INPUTASSET\n";
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/InputSystem/Editor/InputActionAsset/CopyPasteUtility.cs]: Naming of internal consts is kConstant.

File: Packages/com.unity.inputsystem/InputSystem/Editor/Internal/InputActionSerializationHelpers.cs [hunk 3]
```diff
@@ -64,6 +64,30 @@ public static void AddActionMap(SerializedObject asset)
             mapProperty.FindPropertyRelative("m_Bindings").ClearArray();
         }
 
+        public static void AddActionMapFromObject(SerializedObject asset, InputActionMap map)
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/InputSystem/Editor/Internal/InputActionSerializationHelpers.cs:67]: `AddActionMapToActionAssetObject`

File: Packages/com.unity.inputsystem/InputSystem/Editor/Internal/InputActionSerializationHelpers.cs [hunk 4]
```diff
@@ -185,6 +262,47 @@ private static string FindUniqueName(SerializedProperty arrayProperty, string ba
...
+            nameProperty.stringValue = newName;
+        }
+
+        public static void RenameActionMap(SerializedProperty actionMapProperty, string newName)
+        {
+            var nameProperty = actionMapProperty.FindPropertyRelative("m_Name");
+            nameProperty.stringValue = newName;
+        }
+
+        public static void RenameComposite(SerializedProperty actionMapProperty, string newName)
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/InputSystem/Editor/Internal/InputActionSerializationHelpers.cs]: `actionMapProperty` probably copypaste here. What does this function do?

**Review by Rene-Damm — CHANGES_REQUESTED** [review]
Not a single test :(

File: Packages/com.unity.inputsystem/InputSystem/Editor/InputActionAsset/CopyPasteUtility.cs (hunk 1 above)
Comment (tomekpaszek) [reply Packages/com.unity.inputsystem/InputSystem/Editor/InputActionAsset/CopyPasteUtility.cs]: fixed

File: Packages/com.unity.inputsystem/InputSystem/Editor/InputActionAsset/CopyPasteUtility.cs (hunk 2 above)
Comment (tomekpaszek) [reply Packages/com.unity.inputsystem/InputSystem/Editor/InputActionAsset/CopyPasteUtility.cs]: Fixed. Isn't the official standard using "k_" tho?

File: Packages/com.unity.inputsystem/InputSystem/Editor/Internal/InputActionSerializationHelpers.cs (hunk 3 above)
Comment (tomekpaszek) [reply Packages/com.unity.inputsystem/InputSystem/Editor/Internal/InputActionSerializationHelpers.cs:67]: There is "AddActionMap" that simply adds an empty map. "FromObject" means the given object will be added to the map. There is a method "AddActionMap" thus i called this one "AddActionMapFromObject"

File: Packages/com.unity.inputsystem/InputSystem/Editor/Internal/InputActionSerializationHelpers.cs (hunk 4 above)
Comment (tomekpaszek) [reply Packages/com.unity.inputsystem/InputSystem/Editor/Internal/InputActionSerializationHelpers.cs]: It's meant for renaming the composite group. It's how designed the UI for now.

**Pull request merged**